packages/core/src/components/forms/_numeric-input.scss

```diff
@@ -11,27 +14,6 @@
     min-height: 0;
     padding: 0;
     width: $pt-button-height;
-
-    &:first-child {
```

Comment:
removed all logic to unround border-radius for consecutive components in numeric input